framework/src/org/checkerframework/framework/type/AnnotatedTypeFactory.java

```diff
@@ -2411,18 +2432,46 @@ protected void addAliasedAnnotation(String canonicalName, AnnotationMirror type)
     }
 
     /**
+     * Add the annotation whose canonical name is given by {@code canonicalName} as an alias for the
+     * annotation {@code type}. An {@code copyElements} flag is passed in to indicate whether we
+     * want to copy the elements over when getting the alias from the canonical one. An optional
+     * {@code ignorableElements} contains a list of elements that can be safely dropped when the
+     * elements are being copied over.
+     */
+    protected void addAliasedAnnotation(
```

Comment:
The `String` version is provided because in some case the alias class cannot be referenced directly (e.g., the usage of the 2-param `String` version of this method). I still keep this version here since the implementation cannot be moved to the other version without code duplication (cannot get class from string). I've updated the javadoc to explain this)